cpp/src/io/parquet/decode_fixed.cu

```diff
@@ -495,6 +585,123 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size)
   if (t == 0 and s->error != 0) { set_error(s->error, error_code); }
 }
 
+/**
+ * @brief Kernel for computing fixed width non dictionary column data stored in the pages
+ *
+ * This function will write the page data and the page data's validity to the
+ * output specified in the page's column chunk. If necessary, additional
+ * conversion will be performed to translate from the Parquet datatype to
+ * desired output datatype.
+ *
+ * @param pages List of pages
+ * @param chunks List of column chunks
+ * @param min_row Row index to start reading at
+ * @param num_rows Maximum number of rows to read
+ * @param error_code Error code to set if an error is encountered
+ */
+template <typename level_t>
+CUDF_KERNEL void __launch_bounds__(decode_block_size)
+  gpuDecodeSplitPageDataFlat(PageInfo* pages,
+                             device_span<ColumnChunkDesc const> chunks,
+                             size_t min_row,
+                             size_t num_rows,
+                             kernel_error::pointer error_code)
+{
+  __shared__ __align__(16) page_state_s state_g;
+  __shared__ __align__(16) page_state_buffers_s<rolling_buf_size,  // size of nz_idx buffer
+                                                1,                 // unused in this kernel
+                                                1>                 // unused in this kernel
```

Comment:
seems like these can be zeroes, same with line 485

Reply:
The template params are used to size an array, so I'm not sure if it is allowed to be 0.

Reply:
I wasn't sure either, but it seems to work (on my system :smiling_imp:)
Also, some other kernels use zeros for these, e.g. `gpuDecodeDeltaBinary`

Reply:
Hmm...I guess I'll try 0s then 😅

Reply:
yeah, I'm also surprised it works

Reply:
After some offline discussion/research, we've determined `1` is the correct param for unused arrays. C++ does not support 0 length arrays in most instances, but a compiler extension was allowing it here.